homeassistant/components/nobo_hub/climate.py

```diff
@@ -0,0 +1,310 @@
+            await self._switch_profile(self._command_off_id)
+        else:
+            if hvac_mode == HVACMode.AUTO:
+                await self.async_set_preset_mode(PRESET_NONE)
+            elif hvac_mode == HVACMode.HEAT:
+                await self.async_set_preset_mode(PRESET_COMFORT)
+            if self._attr_hvac_mode == HVACMode.OFF:
+                # Turn on by switching to configured "on" profile
+                await self._switch_profile(self._command_on_id)
+        self._attr_hvac_mode = hvac_mode
+
+    async def _switch_profile(self, week_profile_id):
+        """Switch week profile. Used to switch heaters on and off."""
+        await self._nobo.async_update_zone(self._id, week_profile_id=week_profile_id)
+        _LOGGER.debug(
+            "Turned %s heater %s '%s' by switching to week profile %s",
+            "off" if week_profile_id == self._command_off_id else "on",
+            self._id,
+            self._attr_name,
+            self._command_off_id,
+        )
+        # When switching between AUTO and OFF an immediate update does not work (the Nobø API seems to answer with old values), but it works if we add a short delay:
+        await asyncio.sleep(0.5)
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new zone override."""
+        if self._nobo.zones[self._id][ATTR_OVERRIDE_ALLOWED] != "1":
+            return
+        if preset_mode == PRESET_ECO:
+            mode = nobo.API.OVERRIDE_MODE_ECO
+        elif preset_mode == PRESET_AWAY:
+            mode = nobo.API.OVERRIDE_MODE_AWAY
+        elif preset_mode == PRESET_COMFORT:
+            mode = nobo.API.OVERRIDE_MODE_COMFORT
+        else:  # PRESET_NONE
+            mode = nobo.API.OVERRIDE_MODE_NORMAL
+        await self._nobo.async_create_override(
+            mode,
+            self._override_type,
+            nobo.API.OVERRIDE_TARGET_ZONE,
+            self._id,
+        )
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        low, high = None, None
+        if ATTR_TARGET_TEMP_LOW in kwargs:
+            low = int(kwargs.get(ATTR_TARGET_TEMP_LOW))  # type: ignore[arg-type]  # ATTR_TARGET_TEMP_LOW is float if set
+        if ATTR_TARGET_TEMP_HIGH in kwargs:
+            high = int(kwargs.get(ATTR_TARGET_TEMP_HIGH))  # type: ignore[arg-type]  # ATTR_TARGET_TEMP_HIGH is float if set
+        if low is not None:
+            if high is not None:
+                low = min(low, high)
+            elif self.target_temperature_high is not None:
+                low = min(low, int(self.target_temperature_high))
+        elif high is not None and self.target_temperature_low is not None:
+            high = max(high, int(self.target_temperature_low))
+        await self._nobo.async_update_zone(
+            self._id, temp_comfort_c=high, temp_eco_c=low
+        )
+
+    @callback
+    def update(self) -> None:
```

Comment:
It's not allowed to name the function `update` if it should run in the event loop. This is a polling entity that will use `update` to poll for new device data, running that on the executor thread pool. If we want it to run in the event loop we should use a coroutine function named `async_update`.

Reply:
Thanks.

I think I messed up here when I realised that the device itself does not send an update when a zone changes mode based on a schedule stored on the device, hence HA needs to poll for this information. The call itself is handled locally by the library by checking the current time against the retrieved profiles. Any other state change is driven by a callback from the library.

I hope it's OK to do make this combination of polling and pushing. Otherwise the component needs a timer of its own that triggers every minute to push the state, and that seemed unnecessary to me when HA already supports polling periodically.